honeybee_radiance_command/ra_gif.py

```diff
@@ -58,12 +58,18 @@ def to_radiance(self, stdin_input=False):
 
         command_parts = [self.command, self.options.to_radiance()]
         cmd = ' '.join(command_parts)
+
+        if stdin_input and self.input and self.output:
+            cmd = '%s | %s %s' % (self.input, cmd, self.input)
```

Comment:
I'm not sure what are you trying to do here. Since I see this in several files let me write a little bit more about the intend of this design and how it is supposed to be used.

In Radiance it's very common to pipe the output from one command to another command which looks like `command_1 options | command_2 options > final.out` - A real world example in radiance is passing the output of rtrace to rcalc for postprocessing. The command in radiance looks like this.

`rtrace scene.oct < sensors.pts | rcal $1=(0.265*$1+0.67*$2+0.065*$3)*17900 > output.res`

The `stdin_input` option is added to commands to allow these sort of piping from one command to another. If you have the input already assigned then the concept of piping is redundant. Here is an example of how it is used in `honeybee-radiance`. I hope this helps to clarify how the `stdin_input` is supposed to be used.

https://github.com/ladybug-tools/honeybee-radiance/blob/e4791a328e7ff62d97cced90d249c3613fdf4227/honeybee_radiance/cli/raytrace.py#L119-L125

Reply:
Thanks for sharing this. Yes, I am not very clear about stdin_input. It's good to see what you shared but I believe, I already knew this. The command you shared is how piping works in the Linux world. Since, I know this, let me ask some specific questions that might help you help me.

1. in rtrace in [this](https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/rtrace.py#L67-L84) method why there's no condition that shares what happens if stdin_input will be True?

2. If I had to guess the command in case stdin_input is True, it will be
`cmd = '%s > %s' % (cmd, self.output)

Reply:
1. I think [this](https://github.com/ladybug-tools/honeybee-radiance-command/pull/151#discussion_r616908614) is the reason why a condition where stdin_input is true is not added.

2. I think my guess is right